File: app/src/main/java/com/shar2wy/moviesapp/models/moviesRepo/MovieLocalDataSource.java
package com.shar2wy.moviesapp.models.moviesRepo;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Flowable;
import io.realm.Realm;
import io.realm.RealmResults;

/**
 * Created by shar2wy
 * on 4/4/18.
 * Description: description goes here
 */
class MovieLocalDataSource {

    private Realm realm;

    public Flowable<List<Movie>> getMovies() {
        realm = Realm.getDefaultInstance();
        RealmResults<Movie> moviesList = realm.where(Movie.class).findAll();
        List<Movie> list = new ArrayList<>();
        list.addAll(moviesList);

        return Flowable.fromArray(list);
    }

    public void saveMovies(final List<Movie> movies) {
        Realm.getDefaultInstance().executeTransaction(realm -> realm.insertOrUpdate(movies));
    }
}

File: app/src/main/java/com/shar2wy/moviesapp/models/moviesRepo/MovieLocalDataSource.kt
package com.shar2wy.moviesapp.models.moviesRepo

import io.reactivex.Flowable
import io.realm.Realm
import java.util.*

/**
 * Created by shar2wy
 * on 4/4/18.
 * Description: description goes here
 */
internal class MovieLocalDataSource {

    private var realm: Realm = Realm.getDefaultInstance()

    val movies: Flowable<List<Movie>>
        get() {
            val moviesList = realm.where(Movie::class.java).findAll()
            val list = ArrayList<Movie>()
            list.addAll(moviesList)
            return Flowable.fromArray(list)
        }

    fun saveMovies(movies: List<Movie>) {
        Realm.getDefaultInstance().executeTransaction { realm -> realm.insertOrUpdate(movies) }
    }
}
